modin/experimental/core/execution/ray/implementations/pandas_on_ray/io/io.py

```diff
@@ -146,7 +146,7 @@ def read_sql(
         BaseQueryCompiler
             A new query compiler with imported data for further processing.
         """
-        from .sql import is_distributed, get_query_info
+        from modin.experimental.core.io.sql.utils import is_distributed, get_query_info
```

Comment:
LGTM! However `read_sql` still has duplication for Unidist and Ray. Have you looked at whether it is possible to get rid of duplication here as well?

Reply:
`read_sql` makes use of the engine specific things like `PandasOnRayDataframePartition`/`PandasOnUnidistDataframePartition`, `RayWrapper`/`UnidistWrapper`. We do not want the engine to depend on other engines (especially when some of them are not installed in the user env).

Reply:
Wouldn't this approach work?
```python
    build_args = dict(
        frame_partition_cls=PandasOnRayDataframePartition,
        query_compiler_cls=PandasQueryCompiler,
        frame_cls=PandasOnRayDataframe,
    )
    read_csv_glob = type(
        "", (RayWrapper, PandasCSVGlobParser, CSVGlobDispatcher), build_args
    )._read
```

Reply:
Experimental `read_sql` has extended set of parameters ([link](https://github.com/modin-project/modin/blob/8fbe04ff64a1c1658d9ed8317dd64a48b8e61ee0/modin/experimental/core/execution/ray/implementations/pandas_on_ray/io/io.py#L98)). `SQLDispatcher` and the classses called inside it does know nothing how to handle those so I don't think this appoach would work. We would need to add new classes `ExperimentalSQLDispatcher`, `ExperimentalPandasSQLParser` and probably something else instead. I don't think that feature should belong this PR.

Reply:
ok